server.js

```diff
@@ -0,0 +1,133 @@
+const express = require('express');
+const knex = require('knex');
+const helmet = require('helmet');
+const morgan = require('morgan');
+const cors = require('cors');
+const session = require('express-session');
+const allRoutes = require("./allRoutes/restrictedRoutes");
+const KnexSessionStore = require('connect-session-knex')(session);
+
+const knexConfig = require('./knexfile');
+
+const server = express();
+
+const db = knex(knexConfig.development);
+
+const sessionConfig = {
+    name:'NewSession',/// default is sid
+    secret: 'Weird Secret: ergedfghrghbnsftghklnrtqrkntqwtgfdibdrgrk4585985675308gfdhfgkl',
+    cookie: {
+        maxAge: 1 * 24 * 60 * 60 * 1000,
+        secure: false, // only send a cookie over https, should be true in production
+    },
+    httpOnly: true, // js cant touch this cookie.
+    resave: false,
+    saveUninitialized: false,
+    store: new KnexSessionStore({ // creating a tavle in the data baase to store the session an
+        tablename:'sessions',
+        sidfieldname: 'sid',
+        knex: db,
+        createtable: true,
+        clearInterval: 1000 * 60 * 60, // only one camel cased, rest is lower cased.
+    }),
+};
+
+
+
+
+// const db = require('./database/dbConfig.js');
+
+
+
+server.use(express.json());
+server.use(helmet());
+server.use(cors());
+server.use(session(sessionConfig));
+server.use(morgan());
+
+server.use('/api/restricted', allRoutes);
+
+
+
+// Test route :
+
+
+
+server.get('/', (req, res) => {
+    res.send(`API working.\n Sanity Check\n Test Route!`);
+});
+
+
+
+//All Routes
+
+server.post('/api/register', (req, res)=>{
+    const user = req.body;
+    if(user.username && user.password){
+        user.password = bcrypt.hashSync(user.password);
+        db('users')
+        .insert(user)
+        .then(ids=>{
+            res.status(201).json({id: ids[0]});
```

Comment:
While this is fine, there are other options for getting the id out of the array,
such as the [`first`](https://knexjs.org/#Builder-first) method:
```javascript
.insert(user)
.first()
.then(id => {
  res.status(201).json({ id });
// ...
```
or, with [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring):
```javascript
.insert(user)
.then(([ id ]) => {
  res.status(201).json({ id });
// ...
```